ml-dhs-aws/src/main/java/com/marklogic/client/dhs/CSVConverter.java

```diff
@@ -11,52 +11,40 @@
-            .configure(CsvParser.Feature.WRAP_AS_ARRAY, false);
-/*
-      factory = new CsvFactory()
-            .configure(CsvParser.Feature.ALLOW_TRAILING_COMMA, true)
-            .configure(CsvParser.Feature.FAIL_ON_MISSING_COLUMNS, false)
-            .configure(CsvParser.Feature.IGNORE_TRAILING_UNMAPPABLE, false)
-            .configure(CsvParser.Feature.INSERT_NULLS_FOR_MISSING_COLUMNS, false)
-            .configure(CsvParser.Feature.SKIP_EMPTY_LINES, true)
-            .configure(CsvParser.Feature.TRIM_SPACES, true)
-            .configure(CsvParser.Feature.WRAP_AS_ARRAY, false);
-  */
-   }
+	private CsvMapper csvMapper;
+	private CountInputStream countInputStream;
 
-// TODO: in job control file, expose schema control over whether the value is a JSON string, number, or boolean
-   public Iterator<ObjectNode> convertObject(InputStream csvSource) throws IOException {
-      CsvSchema firstLineSchema = CsvSchema.emptySchema().withHeader();
+	public CountInputStream getCountInputStream() {
+		return countInputStream;
+	}
+
+	public void setCountInputStream(CountInputStream countInputStream) {
+		this.countInputStream = countInputStream;
+	}
 
-      Iterator<ObjectNode> objectItr = csvMapper
-            .readerFor(ObjectNode.class)
-            .with(firstLineSchema)
-            .readValues(csvSource);
+	public long getByteCount(){
+		return countInputStream.getByteCount();
+	}
 
-      return objectItr;
-   }
-/* NOTE: Jackson CsvParser doesn't support parsing a CSV row as a JSON string
-   to optimize, may need to implement Iterator<String> or a stream on top of CsvParser
+	public CSVConverter() {
+		csvMapper = new CsvMapper().configure(CsvParser.Feature.ALLOW_TRAILING_COMMA, true)
+				.configure(CsvParser.Feature.FAIL_ON_MISSING_COLUMNS, false)
+				.configure(CsvParser.Feature.IGNORE_TRAILING_UNMAPPABLE, false)
+				.configure(CsvParser.Feature.INSERT_NULLS_FOR_MISSING_COLUMNS, false)
+				.configure(CsvParser.Feature.SKIP_EMPTY_LINES, true).configure(CsvParser.Feature.TRIM_SPACES, true)
+				.configure(CsvParser.Feature.WRAP_AS_ARRAY, false);
 
-   public Iterator<String> convertString(InputStream csvSource) throws IOException {
-      CsvSchema firstLineSchema = CsvSchema.emptySchema().withHeader();
+	}
+
+// TODO: in job control file, expose schema control over whether the value is a JSON string, number, or boolean
+	public Iterator<ObjectNode> convertObject(InputStream csvSource) throws IOException {
+		CountInputStream countInputStream = new CountInputStream(csvSource);
+		setCountInputStream(countInputStream);
+		CsvSchema firstLineSchema = CsvSchema.emptySchema().withHeader();
 
-      CsvParser parser = factory.createParser(csvSource);
-      parser.setSchema(firstLineSchema);
-      Iterator<String> stringItr = parser.readValuesAs(String.class);
+		Iterator<ObjectNode> objectItr = csvMapper.readerFor(ObjectNode.class).with(firstLineSchema)
+				.readValues(csvSource);
```

Comment:
Should readValues() take the countInputStream wrapper instead of csvSource so read requests go through countInputStream?